dts/bindings/gpio/nxp,pcal6416a.yaml

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023 SILA Embedded Solutions GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: PCAL6416A 16-bit I2C-based I/O expander
+
+compatible: "nxp,pcal6416a"
+
+include: [i2c-device.yaml, gpio-controller.yaml]
+
+properties:
+  ngpios:
+    required: true
+    const: 16
```

Comment:
I would suggest leaving this as required but non-constant. That way, anyone implementing a device tree with this part would actively need to specify the number of gpios.

With that, you can use a single `nxp,pcal64xxa.yaml` binding, a single `gpio_pcal64xxa.c` driver, a single `gpio_pcal64xxa.h` header, and a single `Kconfig.pcal64xxa`.

Reply:
I assume the non-constant topic is obsolete. The Kconfigs have been merged now.